lib/dialogs/logout-confirmation/index.tsx

```diff
@@ -1,104 +1,28 @@
 
 import type * as S from '../../state';
-import type * as T from '../../types';
-
-type StateProps = {
-  notes: Map<T.EntityId, T.Note>;
-};
 
 type DispatchProps = {
-  closeDialog: () => any;
-  reallyLogout: () => any;
+  continueAction: () => any;
 };
 
-type Props = StateProps & DispatchProps;
-
-export class LogoutConfirmation extends Component<Props> {
-  exportUnsyncedNotes = () => {
-    const { closeDialog, notes } = this.props;
-
-    exportZipArchive(notes).then(closeDialog);
-  };
-
-  render() {
-    const { closeDialog, notes, reallyLogout } = this.props;
-
-    return (
-      <div className="logoutConfirmation">
-        <Dialog onDone={closeDialog} title="Unsynchronized Notes">
-          <p className="explanation">
-            {notes.size > 0
-              ? 'Logging out will delete any unsynchronized notes.'
-              : 'All notes have synchronized!'}
-          </p>
+type Props = DispatchProps;
 
-          {notes.size > 0 && (
-            <Fragment>
-              <p className="explanation-secondary">
-                Possibly unsynchronized notes
-              </p>
-              <section className="change-list">
-                <ul>
-                  {[...notes.entries()].map(([noteId, note]) => {
-                    const { title, preview } = noteTitleAndPreview(note);
-
-                    return (
-                      <li key={noteId}>
-                        <AttentionIcon />
-                        <span className="note-title">{title}</span>
-                      </li>
-                    );
-                  })}
-                </ul>
-              </section>
-            </Fragment>
-          )}
-          {notes.size > 0 && (
-            <button
-              className="export-unsynchronized"
-              onClick={this.exportUnsyncedNotes}
-            >
-              Export unsynchronized notes
-            </button>
-          )}
-
-          <section className="action-button">
-            <button className="log-out" onClick={reallyLogout}>
-              {notes.size > 0 ? 'Log out' : 'Safely log out'}
-            </button>
-          </section>
-        </Dialog>
-      </div>
-    );
-  }
-}
-
-const mapStateToProps: S.MapState<StateProps> = (state) => {
-  const changes = getUnconfirmedChanges(state);
-  const notes = new Map(
-    changes.notes.map((noteId: T.EntityId) => [
-      noteId,
-      state.data.notes.get(noteId),
-    ])
-  );
-
-  return {
-    notes,
-  };
-};
+const LogoutConfirmation = ({ continueAction }: Props) => (
+  <UnsynchronizedConfirmation
+    actionDescription="Logging out will delete any unsynchronized notes."
+    actionName="Log out"
+    actionSafeName="Safely log out"
+    continueAction={continueAction}
+  />
+);
 
 const mapDispatchToProps: S.MapDispatch<DispatchProps> = {
-  closeDialog,
-  reallyLogout: () => ({ type: 'REALLY_LOGOUT' }),
+  continueAction: actions.simperium.reallyLogOut,
```

Comment:
if we leave in these wrapper components then changing `reallyLogout` to `continueAction` feels like it's a step backwards in indicating what the intention is for the component. it's exposing the internals of the component to the caller of the component because we know that as a _logout_ thing the action it wants is some function that _really logs out_. with `continueAction` I'm not sure what that implies.

Reply:
resolved in https://github.com/Automattic/simplenote-electron/pull/2277/commits/094570fc5044611800546774d72d58d716113af6